chains/evm/calls/utils.go

```diff
@@ -86,7 +86,7 @@ func Transact(client ChainClient, txFabric TxFabric, to *common.Address, data []
 	if err != nil {
 		return common.Hash{}, err
 	}
-	tx := txFabric(n.Uint64(), to, big.NewInt(0), gasLimit, gp, data)
+	tx := txFabric(n.Uint64(), to, value, gasLimit, gp, data)
```

Comment:
I think you should handle the case if `value` is `nil` (maybe set it to zero in this case)

Reply:
and I guess `preRun` is there to prevent user from accidentally sending a 0-value tx if they didn't mean to, so this works